### PR TITLE
[minor_change] Add resource and datasource for mso_template (DCNE-217)

### PR DESCRIPTION
**Pull request by akinross**

### CODE REVIEW

File: mso/resource_mso_template.go [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+	err := ndoTemplate.getTemplate(true)
+	if err != nil {
+		return nil, err
+	}
+	ndoTemplate.SetToSchema(d)
+	log.Println("[DEBUG] MSO Template Resource: Import Completed", d.Id())
+	return []*schema.ResourceData{d}, nil
+}
+
+func getListOfStringsFromSchemaList(d *schema.ResourceData, key string) []string {
```
Comment (sajagana) [inline mso/resource_mso_template.go:465]: Can we move it to utils.go?

File: mso/resource_mso_template.go [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+	if values, ok := d.GetOk(key); ok {
+		return convertToListOfStrings(values.([]interface{}))
+	}
+	return nil
+}
+
+func convertToListOfStrings(values []interface{}) []string {
```
Comment (sajagana) [inline mso/resource_mso_template.go:472]: Can we move it to utils.go?

File: mso/resource_mso_template.go [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+	result := []string{}
+	for _, item := range values {
+		result = append(result, item.(string))
+	}
+	return result
+}
+
+func duplicatesInList(list []string) []string {
```
Comment (sajagana) [inline mso/resource_mso_template.go:480]: Can we move it to utils.go?

**Review by sajagana — CHANGES_REQUESTED** [review]

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by shrsr — DISMISSED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by gmicol — DISMISSED** [review]
LGTM

**Review by shrsr — DISMISSED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by samiib — APPROVED** [review]
LGTM
I created DCNE-261 for the CI.

**Comment by akinross** [comment]
> LGTM I created DCNE-261 for the CI.

perfect thanks!

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**